applications/vanilla/library/Jobs/ExecuteBatchDeleteDiscussion.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * @copyright 2009-2020 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+use Garden\Schema\Schema;
+use Vanilla\Scheduler\Job\JobExecutionStatus;
+use Vanilla\Scheduler\Job\JobPriority;
+
+/**
+ * Execute the discussion delete queue.
+ */
+class ExecuteBatchDeleteDiscussion implements Vanilla\Scheduler\Job\LocalJobInterface {
+
+    /** @var DiscussionModel */
+    private $discussionModel;
+
+    /** @var array */
+    private $message;
+
+    /**
+     * Initial job setup.
+     *
+     * @param DiscussionModel $discussionModel
+     */
+    public function __construct(DiscussionModel $discussionModel) {
+        $this->discussionModel = $discussionModel;
+    }
+
+    /**
+     * Validate the message against the schema.
+     *
+     * @return Schema
+     */
+    private function messageSchema(): Schema {
+        $schema = Schema::parse([
+            "discussionID" => [
+                [
+                    "type" => "integer"
+                ]
+            ]
+        ]);
```

Comment:
Seems like too many nested arrays for this structure. You have the top-level for your fields. The `discussionID` field is defined by an indexed array with an associative array inside it. I don't think that's really defining anything. The schema class will probably not detect the type and allow this field to be any value, regardless of type.